src/api/auth-api.ts

```diff
@@ -0,0 +1,18 @@
+import { AxiosResponse } from 'axios';
+import { apiYandex } from './api';
+import { SignInRequest, UserDTO } from './types';
+import { User } from '../types';
+import { transformUserToDTO } from '../utils/transformUser';
+
+export default {
+  async signin(user: SignInRequest): Promise<void> {
+    await apiYandex.post('/auth/signin', user);
+  },
+
+  async signup(user: User): Promise<AxiosResponse<number>> {
+    const userDTO: UserDTO = transformUserToDTO(user);
+    const response = await apiYandex.post('/auth/signup', userDTO);
+
+    return response?.data?.id;
+  },
```

Comment:
❗️
Не совсем так:
```suggestion
  async signup(user: User): Promise<number> {
    const userDTO: UserDTO = transformUserToDTO(user);
    const response = await apiYandex.post<{ id: number }>('/auth/signup', userDTO);

    return response?.data?.id;
  },
```

Вот что поменялось:
1. Мы возвращаем `id`, на `response`, следовательно возвращаемый промис резолвится в `Promise<number>`, а не в `Promise<AxiosResponse<number>>`
2. Дополнительно типизируем возвращаемое значение ручки `/auth/signup` через дженерик метода `post`

Reply:
А тут не важно что id не response, а в response.data?

Reply:
Это ок, потому что этот дженерик типизирует именно `data`. В этом можно убедиться, провалившись в определение `AxiosResponse` в `index.d.ts` файле либы